src/utils/proposals.js

```diff
@@ -0,0 +1,366 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { v4 } from 'uuid'
+import hathorLib, { PartialTx, PartialTxInputData, PartialTxProposal } from "@hathor/wallet-lib";
+import { DECIMAL_PLACES, TOKEN_MINT_MASK, TOKEN_MELT_MASK, HATHOR_TOKEN_CONFIG } from "@hathor/wallet-lib/lib/constants";
+import { get } from 'lodash';
+import walletUtil from "./wallet";
+
+const { wallet: oldWallet } = hathorLib;
+/**
+ * @typedef ProposalData
+ * @property {string} id Proposal identifier
+ * @property {number} [amountTokens] Derived from proposal data, simplified for exhibition
+ * @property {string} [signatureStatus] Derived from proposal data, simplified for exhibition
+ * @property {boolean} [canISign] Derived from proposal data, simplified for calculations
+ * @property {string} partialTx Serialized unencrypted proposal data
+ * @property {string} [signatures] Merged serialized signatures from all participants
+ * @property {number} [timestamp] Last update timestamp
+ * @property {{partialTx: string, timestamp: number}[]} history History objects array
+ */
+
+/**
+ * @typedef ReduxProposalData
+ * @property {string} id Proposal identifier
+ * @property {string} password Proposal password, stored locally
+ * @property {string} status Saga loading status, from PROPOSAL_DOWNLOAD_STATUS
+ * @property {string} [errorMessage] Error message, present when status is failed
+ * @property {string} [oldStatus] Saga-related information, from PROPOSAL_DOWNLOAD_STATUS
+ * @property {number} [updatedAt] Saga-related last update timestamp
+ * @property {ProposalData} [data] Full decrypted proposal data
+ * @property {BackendProposalData} [rawData] Raw data received from the backend
+ */
+
+/**
+ * @typedef BackendProposalData
+ * @property {string} id
+ * @property {string} partialTx Encrypted PartialTx data
+ * @property {string} signatures Encrypted Signatures data
+ * @property {number} timestamp Latest update timestamp
+ * @property {{partialTx: string, timestamp: number}[]} history Historic data
+ */
+
+/**
+ * Generates a random positive integer between the maximum and minimum values,
+ * with the default minimum equals zero
+ * @param {number} max
+ * @param {number} [min=0]
+ * @returns {number} Random number
+ */
+export function getRandomInt(max, min = 0) {
```

Comment:
Why do you need this method? We have `lodash` in the wallet, which has exactly this method: https://lodash.com/docs/4.17.15#random

Reply:
Actually we don't! :eyes: 
Lodash is not on our current dependencies, and is only available to our code through other dependencies' dependencies.
Should I add this?

And this `getRandomInt` function is only used on this PR's mocked HTTP responses on saga methods. It will be discarded as soon as the integration with the backend happens.

Reply:
We use `lodash` in many files, e.g. https://github.com/HathorNetwork/hathor-wallet/blob/master/src/utils/wallet.js#L38

We should change to use the `lodash` method, so we don't forget to remove this in the future.

Reply:
This function ended up being removed on [1402ad7](https://github.com/HathorNetwork/hathor-wallet/pull/361/commits/1402ad7b3d224c11d2ae5e6aea24a183527a5717) due to some code suggestions from Carneiro.

About lodash, on this file you mention I'm receiving the following warning on my IDE: `Module is not listed in package.json dependencies`

I think as soon as we add a linter to the project this will have to be fixed.